core/store/src/metrics.rs

```diff
@@ -214,3 +214,81 @@ pub static COLD_MIGRATION_READS: Lazy<IntCounterVec> = Lazy::new(|| {
     )
     .unwrap()
 });
+
+pub static FLAT_STORAGE_HEAD_HEIGHT: Lazy<IntGaugeVec> = Lazy::new(|| {
+    try_create_int_gauge_vec(
+        "flat_storage_head_height",
+        "Height of flat storage head",
+        &["shard_id"],
+    )
+    .unwrap()
+});
+pub static FLAT_STORAGE_CACHED_BLOCKS: Lazy<IntGaugeVec> = Lazy::new(|| {
+    try_create_int_gauge_vec(
+        "flat_storage_cached_blocks",
+        "Number of cached blocks in flat storage",
```

Comment:
Maybe also name this as CACHED_DELTAS to be consistent with the other metrics here?

Reply:
Actually we need both, because logic for adding deltas and blocks is _slightly_ different. Added